ADC_DMA/Core/Src/main.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * 	Implement a routine that reads the internal voltage reference (ADC_IN17)
+ * 	and internal temperature sensor (ADC_IN18).
+ * 	The internal reference voltage must be in millivolts and the temperature in Celsius.
+ *
+ * 	1 - Polling		[X]
+ * 	2 - Interrupt	[X]
+ * 	3 - DMA			[/]
+ */
+
+#include "stm32l053xx.h"
+#include "macros.h"
+
+typedef struct {
+	uint16_t	v_ref;			//internal reference voltage measured (after calculation)
+	int16_t		temperature;	//temperature in Celsius degrees measured (after calculation)
+}sADC_measure;	//struct with all the information read by the ADC channel
+
+void 		wait(uint16_t time);	//delay function, system clock based
+
+int main(void)
+{
+	sADC_measure 	ADC_measure = {0};	//struct with all the information read by the ADC channel
+	uint16_t 		aux = 0;			//we only use aux to calculate v_ref
+
+	/*							  *
+	 * --- DMA INITIALIZATION --- *
+	 *							  */
+	RCC->AHBENR |= RCC_AHBENR_DMAEN;
```

Comment:
RM indicates the steps that we need to follow in order to init DMA. I would recommend to follow those because of good practices.
![image](https://user-images.githubusercontent.com/25661168/83529963-76fa9b00-a4c1-11ea-81c2-4671b6af8921.png)